compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConv2DToWinograd.cpp

```diff
@@ -291,28 +165,60 @@ class ConvertConvToWinograd final : public OpRewritePattern<ConvOp> {
   LogicalResult matchAndRewrite(ConvOp convOp,
                                 PatternRewriter &rewriter) const override {
 
-    bool isNchw;
-    if (!isValidConv2d(convOp, isNchw)) {
+    bool isNchwFchw;
+    if (!isValidConv2d(convOp, isNchwFchw)) {
       return failure();
     }
 
-    // Check that kernel has been constant folded (by validating rank = 3)
+    // Create winograd filter transform op.
     Value kernel = convOp.getInputs()[1];
     auto kernelType = cast<ShapedType>(kernel.getType());
     if (!kernelType) {
       return failure();
     }
-    Type elementType = kernelType.getElementType();
-    ArrayRef<int64_t> kernelShape = kernelType.getShape();
-    if (kernelShape.size() != 3) {
-      return failure();
+    if (!kernelType.hasStaticShape()) {
+      return rewriter.notifyMatchFailure(convOp->getLoc(),
+                                         "Kernel shape is not static");
     }
+    SmallVector<int64_t> kernelShape(kernelType.getShape());
+    const int64_t kh = isNchwFchw ? kernelShape[2] : kernelShape[0];
+    const int64_t kw = isNchwFchw ? kernelShape[3] : kernelShape[1];
+    if ((kh != 3) || (kw != 3)) {
+      return rewriter.notifyMatchFailure(convOp->getLoc(),
+                                         "Winograd only supports 3x3 filters");
+    }
+    assert(kernelShape.size() == 4);
+    Type elementType = kernelType.getElementType();
 
     const int64_t kernelSize = 3;
     const int64_t inputTileSize = outputTileSize + kernelSize - 1;
 
-    // Create winograd input transform op
     Location loc = convOp.getLoc();
+    const std::array<int64_t, 2> hwcfKernelDims{0, 1};
+    const std::array<int64_t, 2> fchwKernelDims{2, 3};
```

Comment:
Prefer using `()` constructor or `=` assignment instead of brace-list initialization. This kind of code is dangerous when the type is not an integer type. E.g.,

```
std::vector<std::string> strings{2}; // A vector of two empty strings.
std::vector<int> ints{2};            // A vector containing only the integer 2.
```

Even it works for this case, I'd suggest to use either direct constructer or assignment style for consistency in the codebase.

See https://abseil.io/tips/88 for more details.